scripts/misc/download_hf_model.py

```diff
@@ -0,0 +1,248 @@
+        save_dir (str, optional): The path to the directory where the model files will be downloaded. If `None`, reads
+            from the `HUGGINGFACE_HUB_CACHE` environment variable or uses the default Hugging Face Hub cache directory.
+        prefer_safetensors (bool): Whether to prefer Safetensors weights over PyTorch weights if both are
+            available. Defaults to True.
+        token (str, optional): The HuggingFace API token. If not provided, the token will be read from the
+            `HUGGING_FACE_HUB_TOKEN` environment variable.
+    """
+    repo_files = set(hf_hub.list_repo_files(repo_id))
+
+    # Ignore TensorFlow, TensorFlow 2, and Flax weights as they are not supported by Composer.
+    ignore_patterns = [
+        "*.ckpt",
+        "*.h5",
+        "*.msgpack",
+    ]
+
+    if (
+        SAFETENSORS_WEIGHTS_NAME in repo_files
+        or SAFETENSORS_WEIGHTS_INDEX_NAME in repo_files
+    ) and prefer_safetensors:
+        print("Safetensors found and preferred. Excluding pytorch files")
+        ignore_patterns.append(PYTORCH_WEIGHTS_PATTERN)
+    elif PYTORCH_WEIGHTS_NAME in repo_files or PYTORCH_WEIGHTS_INDEX_NAME in repo_files:
+        print(
+            "Safetensors not found or prefer_safetensors is False. Excluding safetensors files"
+        )
+        ignore_patterns.append(SAFETENSORS_EXCLUDE_PATTERN)
+    else:
+        raise ValueError(
+            f"No supported model weights found in repo {repo_id}."
+            + " Please make sure the repo contains either safetensors or pytorch weights."
+        )
+
+    download_start = time.time()
+    hf_hub.snapshot_download(
+        repo_id, cache_dir=save_dir, ignore_patterns=ignore_patterns, token=token
+    )
+    download_duration = time.time() - download_start
+    print(
+        f"Downloaded model {repo_id} from Hugging Face Hub in {download_duration} seconds"
+    )
+
+
+def _extract_links_from_html(html: str):
+    """Extracts links from HTML content.
+
+    Args:
+        html (str): The HTML content
+
+    Returns:
+        list[str]: A list of links to download.
+    """
+    soup = BeautifulSoup(html, "html.parser")
+    links = [a["href"] for a in soup.find_all("a")]
+    return links
+
+
+def _recursive_download(
+    session: requests.Session,
+    base_url: str,
+    path: str,
+    save_dir: str,
+    ignore_cert: bool = False,
+):
+    """Downloads all files from a directory on a remote server, including subdirectories.
+
+    Args:
+        session: A requests.Session through which to make requests to the remote server.
+        url (str): The base URL where the files are located.
+        path (str): The path from the base URL to the files to download. The full URL for the download is equal to
+            '<base_url>/<path>'.
+        save_dir (str): The directory to save downloaded files to.
+        ignore_cert (bool): Whether or not to ignore the validity of the SSL certificate of the remote server.
```

Comment:
whats the purpose of even having this as an option?

Reply:
If you want to host your own cache server with a self-signed TLS certificate, you'll need to specify this option